pandas/tests/io/test_sql.py

```diff
@@ -226,10 +226,8 @@ def _get_all_tables(self):
     def _close_conn(self):
         from pymysql.err import Error
 
-        try:
+        with pytest.raises(Error):
             self.conn.close()
-        except Error:
-            pass
```

Comment:
I think this is kind of a "teardown" method of the tests, so not sure this should use an assert instead of try/except